src/Microsoft.AspNet.Mvc.Core/Controller.cs

```diff
@@ -1209,6 +1209,8 @@ public virtual bool TryValidateModel([NotNull] object model, string prefix)
                 throw new InvalidOperationException(message);
             }
 
+            ModelStateDictionaryUtility.ClearModelStateDictionary(ModelState, model.GetType());
```

Comment:
Yes I am just clearing the entries for properties in model. Will use a better name for the method.